### ISSUE TITLE
Gatsby Segaults on M1 Mac with `gatsby-config.ts` file

### ISSUE THREAD

**Issue by knpwrs** [gatsbyjs/gatsby#35064]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

When using a `gatsby-config.ts` file instead of a `gatsby-config.js` file on an M1 MacBook Pro, Gatsby Segfaults with no useful error messages.

This appears to be a bug in Parcel, and is fixed here (Gatsby may need an upgrade?): https://github.com/parcel-bundler/parcel/issues/7702

A workaround for the time being is to set `PARCEL_WORKERS=0`, e.g., `"develop": "PARCEL_WORKERS=0 gatsby develop"`.

### Reproduction Link

https://github.com/knpwrs/gatsby-ts-segfault

### Steps to Reproduce

1. Use a MacBook Pro with an M1 / Apple Silicon
2. Run `npm init gatsby` and use TypeScript
3. Rename `gatsby-config.js` to `gatsby-config.ts`
4. Run `npm run develop`
5. Segfault, no useful errors

### Expected Result

Gatsby shouldn't segfault.

### Actual Result

Gatsby segfaults.

### Environment

```shell
System:
    OS: macOS 12.1
    CPU: (10) arm64 Apple M1 Pro
    Shell: 5.8 - /opt/homebrew/bin/zsh
  Binaries:
    Node: 17.3.1 - /opt/homebrew/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.5.1 - /opt/homebrew/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 98.0.4758.109
    Firefox: 97.0
    Safari: 15.2
  npmPackages:
    gatsby: ^4.9.0 => 4.9.1 
    gatsby-plugin-manifest: ^4.9.0 => 4.9.0 
    gatsby-plugin-mdx: ^3.9.0 => 3.9.0 
    gatsby-plugin-preact: ^6.9.0 => 6.9.0 
    gatsby-plugin-react-helmet: ^5.9.0 => 5.9.0 
    gatsby-plugin-sitemap: ^5.9.0 => 5.9.0 
    gatsby-remark-prismjs: ^6.9.0 => 6.9.0 
    gatsby-source-filesystem: ^4.9.0 => 4.9.0
```

**Comment by knpwrs** [comment]
I believe if this gets bumped to `2.3.2` this issue should be resolved: https://github.com/gatsbyjs/gatsby/blob/e7d08e20c816752c2754771f87fb94da33d9cbb7/packages/gatsby/package.json#L24

**Comment by LekoArts** [comment]
Hi! You're able to fix this by deleting your lock file and making sure that Parcel installs lmdb later than 2.2.1 - we're unpinning the version additionally, then this shouldn't occur. But you can already fix this yourself in the meantime.

**Comment by LekoArts** [comment]
This should be fixed in `gatsby@4.9.2` inside Gatsby (Parcel pins to 2.2.3 at the moment)
